docs/core/integration-quality-scale/rules/iqs005.md

```diff
@@ -0,0 +1,36 @@
+---
+title: "IQS005: Support config entry unloading"
+---
+
+IQS005 is as follows:
+
+Support config entry unloading
+
+## Reasoning
+Sometimes a config entry ends up in a state where it is stuck.
+This can have many reasons, but sometimes the integration works again after the integration has been reloaded.
+
+In order to support this, the integration should support unloading the config entry.
+This gives users the ability to reload the integration without having to restart Home Assistant.
+Reloading an integration can be a powerful tool for users to fix issues with integrations.
+
+## Example implementation
+The code required to reload an integration is very little, with a lot of benefits for the user.
+
+The method that has to be added to `__init__.py` looks very similar to the `async_setup_entry` method.
+
+`__init__.py`:
+```python
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+```
+
+## Additional resources
+More information about config entries and their lifecycle can be found in the [config entry documentation](../../../config_entries_index).
+
+## Exceptions
+There are no exceptions to this rule.
```

Comment:
I am wondering if we make an exception for integrations that just do not need a reload. I can't think of any from the top of my head right now, but there are some that for example only provide services

Reply:
I cannot think of an exception tbh.